stdpopsim/species.py

```diff
@@ -292,7 +293,21 @@ def get_demographic_model(self, id):
         for model in self.demographic_models:
             if model.id == id:
                 return model
-        raise ValueError(f"DemographicModel '{self.id}/{id}' not in catalog")
+        suggestions = difflib.get_close_matches(  # Can't harm to use a liberal cutoff
```

Comment:
Let's make a function for this in ``util`` as you suggest: ``get_suggestions(query, possibilities)`` which returns the list of suggestions, and write some test cases for that. We can then make another function which takes the results of this and formats a nice error string from this, which we can then reuse to make error messages anywhere we're doing ID lookup like this.

We should do some tests on some hard-coded examples from the catalog to make sure it does what we expect, too.